ndk/src/input_queue.rs

```diff
@@ -71,14 +71,14 @@ impl InputQueue {
         }
     }
 
-    pub fn attach_looper(&self, looper: &ForeignLooper, id: u32) {
+    pub fn attach_looper(&self, looper: &ForeignLooper, id: i32) {
```

Comment:
We should probably add userdata as argument to this function already?

Reply:
you really can't pass user data without making attach_looper unsafe I don't think.

```rust
pub fn attach_looper<'a, T>(&'a self, looper: &ForeignLooper, id: i32, &'a T)
```

looking at the code the lifetime of looper isn't correct, so maybe it should be marked unsafe anyway

Reply:
Hmm yeah it doesn't look like the lifetime generic of `RwLockReadGuard` is propagated in `Deref` so we can't really enforce it to be `'static` either. Might hold off adding this until someone needs it and feels like implementing it :)